frontend/src/Component/EditComponents/EditComponents.tsx

```diff
@@ -117,6 +71,7 @@ export default function EditComponents(props: EditComponentsProp) {
                     </Box>
                     <Button variant={"contained"} onClick={handleSubmitNewComponent} sx={{mt: 1}}
                             disabled={props.components.filter(c => c.category === newComponentCategory).length!==0
+                                || !/^\d+$/.test(newComponentAgeValue.trim())
                                 || newComponentModel===""
                                 || newComponentCategory===""}>
```

Comment:
This gets a bit hard to read. Maybe you could define a `newComponentButtonIsDisabled` constant in the component header and put the logic there? Also, maybe it is a bit shorter to use `.find... !== undefined` instead of `.filter... .length !== 0`?